integration/devmode/devmode_test.go

```diff
@@ -0,0 +1,181 @@
+		chaincodeProcess         ifrit.Process
+		channelName, chaincodeID string
+	)
+
+	BeforeEach(func() {
+		var err error
+		channelName = "testchannel"
+		testDir, err = ioutil.TempDir("", "devmode")
+		Expect(err).NotTo(HaveOccurred())
+
+		client, err = docker.NewClientFromEnv()
+		Expect(err).NotTo(HaveOccurred())
+
+		chaincode = nwo.Chaincode{
+			Name:    "mycc",
+			Version: "0.0",
+			Path:    "github.com/hyperledger/fabric/integration/chaincode/simple/cmd",
+			Ctor:    `{"Args":["init","a","100","b","200"]}`,
+			Policy:  `AND ('Org1MSP.member','Org2MSP.member')`,
+		}
+
+		chaincodeID = fmt.Sprintf("%s:%s", chaincode.Name, chaincode.Version)
+		network = nwo.New(nwo.BasicSolo(), testDir, client, BasePort(), components)
+		network.TLSEnabled = false
+		// turn on dev mode for all peers
+		for _, p := range network.Peers {
+			p.DevMode = true
+		}
+
+		network.GenerateConfigTree()
+		network.Bootstrap()
+
+		networkRunner := network.NetworkGroupRunner()
+		process = ifrit.Invoke(networkRunner)
+		Eventually(process.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+
+	AfterEach(func() {
+		if process != nil {
+			process.Signal(syscall.SIGTERM)
+			Eventually(process.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+
+		if chaincodeProcess != nil {
+			chaincodeProcess.Signal(syscall.SIGTERM)
+		}
+
+		if network != nil {
+			network.Cleanup()
+		}
+
+		os.RemoveAll(testDir)
+	})
+
+	It("executes chaincode in dev mode", func() {
+		org1peer0 := network.Peer("Org1", "peer0")
+		orderer := network.Orderer("orderer")
+
+		By("setting up the channel")
+		network.CreateAndJoinChannel(orderer, channelName)
+
+		By("building the chaincode")
+		chaincodePath, err := gexec.Build(chaincode.Path)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("running the chaincode")
+		peerChaincodeAddress := network.PeerAddress(org1peer0, nwo.ChaincodePort)
+		envs := []string{fmt.Sprintf("CORE_PEER_ADDRESS=%s", peerChaincodeAddress), fmt.Sprintf("CORE_CHAINCODE_ID_NAME=%s", chaincodeID)}
+		cmd := exec.Command(chaincodePath)
+		cmd.Env = append(cmd.Env, envs...)
+		chaincodeRunner = ginkgomon.New(ginkgomon.Config{
+			Name:              "chaincode",
+			Command:           cmd,
+			StartCheck:        `starting up ...`,
+			StartCheckTimeout: 15 * time.Second,
+		})
+		chaincodeProcess = ifrit.Invoke(chaincodeRunner)
+		Eventually(chaincodeProcess.Ready(), network.EventuallyTimeout).Should(BeClosed())
+
+		By("installing the chaincode")
+		peers := network.PeersWithChannel(channelName)
+		nwo.InstallChaincode(network, chaincode, peers...)
+		By("Instantiating the chaincode")
+		nwo.InstantiateChaincode(network, channelName, orderer, chaincode, org1peer0, peers...)
+		By("Querying and invoking the chaincode")
+		RunQueryInvokeQuery(network, orderer, org1peer0, channelName, 100)
+
+		By("killing the chaincode process")
+		chaincodeProcess.Signal(syscall.SIGKILL)
+		Eventually(chaincodeProcess.Wait(), network.EventuallyTimeout).Should(Receive())
+
```

Comment:
This seems like a good place to address Matt's comment "What asserts that our "dev" chaincode is being used?" by attempting an invoke of the chaincode and getting a failure since we stopped the chaincodef.

Reply:
🧹 See updates